collections/_posts/2022-07-06-Vision_Transformer_Network_for_Image_Anomaly_Detection_and_Localization.md

```diff
@@ -0,0 +1,116 @@
+    venue:   "2021 IEEE 30th International Symposium on Industrial Electronics (ISIE)"
+pdf: "https://arxiv.org/pdf/2104.10036.pdf"
+---
+
+
+# Context
+
+* **Anomaly detection**, also called *outlier detection* or *one-class classification*, is the process of detecting events/items that deviate significantly from the normality. A distribution of the *normal class* is estimated and the event/item is said to be anomalous if does not belong to this distribution.
+* Here we tackle the problem of anomaly detection in images in the context of machine learning, meaning we will train a network (or another kind of algorithm) on *normal* images only, and then, at inference, we will try to classify images as normal VS anomalous.
+![](/collections//images/VT-ADL/mvtec_ad.jpg)  
+*Example of normal (green) VS anomalous (red) image on the MVTecAD dataset*
+* The objective of this paper is to propose an anomaly detection method which make use of a **transformer**, a **decoder** and a Gaussian Mixture Model (**GMM**) **estimation network**.
+* The authors also introduce a database similar to MVTecAD, containing 3 objects named BTAD.
+
+
+# Highlights
+
+* Combination of **transformer**, **decoder** and **GMM** for anomaly detection.
+* Introduction of a **new "MVTec like" dataset** named **BTAD**.
+
+
+# Methods
+
+The model is divided in 3 blocks :
+
+### 1) Transformer-encoder
+#### 1.a) Transformer
+
+The transformer architecture is the same as in the original ViT paper. The image is decomposed in patches, embedded through a linear layer, summed with a position embedding and then passed through multiple encoding layers.
+
+$$ \mathbb{R}^{H \times W \times C} \xrightarrow[\text{division}]{\text{patch}}  \mathbb{R}^{N \times P \times P \times C} \xrightarrow[\text{layer}]{\text{linear}} \mathbb{R}^{N \times D} \xrightarrow[\text{}]{\text{transformer}} \mathbb{R}^{N \times D} $$  
+
+With $$H \times W$$ the size of the base image, $$C$$ the channel size, $$P \times P$$ the patch size and $$N$$ the number of patch.  
+
+**The reduction operation is thus done in the linear projection layer**.
+
+![](/collections/images/VT-ADL/fig2_short.jpg)
+
+
+<sup> Note : In the paper the authors mention a dimension $$(N+1)$$ as if there is a class token, but I saw no use of a class token, nor do they speak about it, so I think the dimension is $$N$$ and not $$(N+1)$$. </sup>   
+
+#### 1.b) Encoder
+
+After the transformer, the output sequence is reduced in dimension again before being fed to the decoder. The authors used a linear projection step. They also mention a sum of the sequence, so both might be feasible and operate the same dimension reduction.
+
+$$ \mathbb{R}^{N \times D} \xrightarrow[\text{projection}]{\text{linear}} \mathbb{R}^{D}$$
+
+### 2) Decoder
+
+The decoder is a **classical CNN decoder**, consisting of 5 transposed convolutional layers, with batch-normalization and ReLu activation, except for the last layer which uses tanh activation.
```

Comment:
I think it is not classic to decode a vector (1 dim) into an image (3 dim) with convolutions, maybe there is a missing step you can precise.

Reply:
added this precision